op-node/p2p/sync.go

```diff
@@ -620,14 +620,20 @@ func (s *SyncClient) doRequest(ctx context.Context, id peer.ID, n uint64) error
 	if err != nil {
 		return fmt.Errorf("failed to read response: %w", err)
 	}
+
+	blockVersion := eth.BlockV1
+	if s.cfg.IsCanyon(expectedBlockNum) {
```

Comment:
The activation is done by timestamp, not block number

Reply:
Good catch, thanks. I've just added a function to go from blockNumber to blockTimestamp on the config. `TimestampForBlock`. 

The specs lead me to believe that this should be deterministic. Would appreciate a sanity check on this, as I maybe e misunderstanding.

Reply:
It is deterministic. Code looks correct. I skimmed it & there doesn't seem to be an easy way to introduce this otherwise.